dao/src/test/java/org/thingsboard/server/dao/service/timeseries/nosql/TimeseriesServiceNoSqlSetNullEnabledTest.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright © 2016-2024 The Thingsboard Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.thingsboard.server.dao.service.timeseries.nosql;
+
+import com.datastax.oss.driver.api.core.uuid.Uuids;
+import org.junit.Test;
+import org.springframework.test.context.TestPropertySource;
+import org.thingsboard.server.common.data.id.DeviceId;
+import org.thingsboard.server.common.data.kv.Aggregation;
+import org.thingsboard.server.common.data.kv.BaseReadTsKvQuery;
+import org.thingsboard.server.common.data.kv.BasicTsKvEntry;
+import org.thingsboard.server.common.data.kv.DoubleDataEntry;
+import org.thingsboard.server.common.data.kv.LongDataEntry;
+import org.thingsboard.server.common.data.kv.TsKvEntry;
+import org.thingsboard.server.dao.service.DaoNoSqlTest;
+import org.thingsboard.server.dao.service.timeseries.BaseTimeseriesServiceTest;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@DaoNoSqlTest
+@TestPropertySource(properties = {
+        "cassandra.query.set_null_values_enabled=true",
+})
+public class TimeseriesServiceNoSqlSetNullEnabledTest extends BaseTimeseriesServiceTest {
+
+    @Test
+    public void shouldSetNullValuesForNonTargetDataTypeColumn() throws ExecutionException, InterruptedException, TimeoutException {
+        long ts = 5000;
+        TsKvEntry entry = new BasicTsKvEntry(ts, new LongDataEntry("temp", 0L));
+        TsKvEntry entry2 = new BasicTsKvEntry(ts, new DoubleDataEntry("temp", 20.6));
+        DeviceId deviceId = new DeviceId(Uuids.timeBased());
+        tsService.save(tenantId, deviceId, entry).get(MAX_TIMEOUT, TimeUnit.SECONDS);
+        tsService.save(tenantId, deviceId, entry2).get(MAX_TIMEOUT, TimeUnit.SECONDS);
+
+        List<TsKvEntry> list = tsService.findAll(tenantId, deviceId, Collections.singletonList(new BaseReadTsKvQuery("temp", 0L,
```

Comment:
It would be nice to have a test with no aggregation and ensure that the entity is exactly ONE and equals the latest one.
It also would be nice to add another data point to have more than 1 point in time-series (ensure that ts is working as expected)